hermeto/core/package_managers/cargo/main.py

```diff
@@ -137,13 +137,27 @@ def _hidden_cargo_config_file(package_dir: RootedPath) -> Generator[None, None,
     """
     do_nothing = None
 
-    config = package_dir.join_within_root(".cargo/config.toml")
-    data = config.path.read_text() if config.path.exists() else None
-    config.path.unlink() if data is not None else do_nothing
+    # There is a slim chance to find an old project with .cargo/config
+    # instead of .cargo/config.toml. If found it has to be hidden too since it still
+    # takes precedence over the now standard .cargo/config.toml
+    # (https://doc.rust-lang.org/cargo/reference/config.html).
+    # Note, that ordering matters here, since .cargo/config could be a symlink
+    # to .cargo/config.toml for projects that are built with both old and new versions
+    # of Cargo. Unlinking a symlink first is safe.
+    all_possible_config_names = (".cargo/config", ".cargo/config.toml")
+    configs_contents = []
+
+    for cfgname in all_possible_config_names:
+        config = package_dir.join_within_root(cfgname)
+        data = config.path.read_text() if config.path.exists() else None
+        configs_contents.append((config, data))
+        config.path.unlink() if data is not None else do_nothing
```

Comment:
nitpick (readability improvement): Pre-existing, but since you're touching the code this compound statement should be written as multiline instead of a one-liner. It's one thing to do this with assignments, but statements are harder to read this way and I'd argue it feels like unnecessary obfuscation. As a bonus, we'll be able to drop the `do_nothing` variable then as it's only needed to satisfy the one-liner condition syntax.

Reply:
Ok, I find both styles quite readable with slight preference to one-liners, but won't insist on it.